src/Panel/Ui/FilePreview.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace Kirby\Panel\Ui;
+
+use Kirby\Cms\App;
+use Kirby\Cms\File;
+use Kirby\Panel\Ui\FilePreviews\FileDefaultPreview;
+use Kirby\Toolkit\I18n;
+
+/**
+ * Defines a component that implements a file preview
+ *
+ * @package   Kirby Panel
+ * @author    Nico Hoffmann <nico@getkirby.com>
+ * @link      https://getkirby.com
+ * @copyright Bastian Allgeier
+ * @license   https://getkirby.com/license
+ * @since     5.0.0
+ * @internal
+ */
+abstract class FilePreview extends Component
+{
+	public string $component = 'k-file-default-preview';
```

Comment:
`Comment:` For these, it's actually a lot easier to just define it on each class on top than handing it through every constructor.

Reply:
Would that be something thaat we should also apply to the Buttons?